src/main/kotlin/telegram/Menu.kt

```diff
@@ -195,124 +284,375 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
     )
 }
 
-internal fun showAdminListMenu(chatId: Long, messageId: Long, bot: Bot) {
-    bot.editMessageReplyMarkup(
-        ChatId.fromId(chatId),
+internal fun showAdminListMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    pageIndex: Int = 0
+) {
+    showPaginatedAdminSubmenu(
+        chatId,
+        messageId,
+        bot,
+        "Список администраторов",
+        admins,
+        {
+            accounts.get(it.chatId)?.let { acc ->
+                row {
+                    button(blankCommand named acc.fullName())
+                    button(removeAdminCommand, it.chatId, messageId, pageIndex)
+                }
+            }
+        },
+        adminSettingsCommand,
+        pageIndex
+    )
+}
+
+internal fun showGameStatusMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    pageIndex: Int = 0
+) {
+    showPaginatedAdminSubmenu(
+        chatId,
         messageId,
+        bot,
+        "Активные игры",
+        games,
+        {
+            button(blankCommand named it.name())
+            button(terminateGameCommand, it.id, messageId)
+        },
+        gamesSettingsCommand,
+        pageIndex
+    )
+}
+
+internal fun <T: Any> showPaginatedMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    title: String,
+    list: List<T>,
+    actionForEach: KeyboardContext.(T) -> Unit,
+    bottomContent: KeyboardContext.(Long) -> Unit,
+    menuCommand: Command,
+    desiredPageIndex: Int,
+    itemsPerPage: Int = defaultItemsPerPage
+) {
+    val msgId = if (messageId == -1L) {
+        sendMessage(
+            bot,
+            chatId,
+            title
+        )
+    } else {
+        messageId
+    }
+    updateMessage(
+        bot,
+        chatId,
+        msgId,
         replyMarkup = inlineKeyboard {
-            button(blankCommand named "Список администраторов")
-            admins.find().forEach {
-                accounts.get(it.chatId)?.let { acc ->
+            button(blankCommand named title)
+            val listSize = list.size
+            if (listSize > 0) {
+                val quotient = listSize / itemsPerPage
+                val totalAvailablePages = if (listSize % itemsPerPage == 0) {
+                    quotient
+                } else {
+                    quotient + 1
+                }
+                val pageIndex = if (desiredPageIndex >= totalAvailablePages || desiredPageIndex < 0) {
```

Comment:
Just take a `desiredPageIndex % totalAvailablePages` and it will auto roll over
Put a rolled over value into your commands it will always point to a valid page